climakitae/core/data_interface.py

```diff
@@ -1577,3 +1586,427 @@ def _display_select(self):
+        Default to None
+    timescale: str, optional
+        Default to None
+    scenario: str, optional
+        Default to None
+    tidy: boolean, optional
+        Format the pandas dataframe? This creates a DataFrame with a MultiIndex that makes it easier to parse the options.
+        Default to True
+
+    Returns
+    -------
+    cat_subset: pd.DataFrame
+        Catalog options for user-provided inputs
+
+    """
+
+    # Get intake catalog and variable descriptions from DataInterface object
+    data_interface = DataInterface()
+    var_pd = data_interface.variable_descriptions
+    catalog = data_interface.data_catalog
+
+    cat_df = _get_user_friendly_catalog(
+        intake_catalog=catalog, variable_descriptions=var_pd
+    )
+
+    # Raise error for bad input from user
+    for user_input in [variable, downscaling_method, resolution, timescale, scenario]:
+        if (user_input is not None) and (type(user_input) != str):
+            print("Function arguments require a single string value for your inputs")
+            return None
+
+    d = {
+        "variable": variable,
+        "timescale": timescale,
+        "downscaling_method": downscaling_method,
+        "scenario": scenario,
+        "resolution": resolution,
+    }
+
+    for key, val in zip(
+        d.keys(), d.values()
+    ):  # Loop through each key, value pair in the dictionary
+        # Use the catalog to find the valid values in the list
+        valid_options = np.unique(cat_df[key].values)
+        if (
+            val == None
+        ):  # If the user didn't input anything for that key, set the values to all the valid options
+            d[key] = valid_options
+            continue  # Don't finish the loop
+        # If the input value is not in the valid options, see if you can help the user out
+        elif val not in valid_options:
+
+            # Perhaps they didn't capitalize it correctly?
+            if val.lower().capitalize() in valid_options:
+                d[key] = [val.lower().capitalize()]
+                continue  # Don't finish the loop
+
+            # This catches any common bad inputs for resolution: i.e. "3KM" or "3km" instead of "3 km"
+            if key == "resolution":
+                try:
+                    good_resolution_input = val.lower().split("km")[0] + " km"
+                    if good_resolution_input in valid_options:
+                        d[key] = [good_resolution_input]
+                        continue
+                except:
+                    pass
+
+            print("Input " + key + "='" + val + "' is not a valid option.")
+            # Use difflib package to make a guess for what the input might have been
+            # For example, if they input "statistikal" instead of "Statistical", difflib will find "Statistical"
+            # Change the cutoff to increase/decrease the flexibility of the function
+            closest_option = difflib.get_close_matches(val, valid_options, cutoff=0.59)
```

Comment:
Awesome! :)

Reply:
+1

Reply:
Checking whether the user input is a substring of any `valid_options` would also be a quick value add, especially when the variable names are so long that it’s hard for `difflib.get_close_matches` to consider them to be similar to any user input that's short. For example, we can then suggest 'West-East component of Wind at 10m' to users interested in “wind”, even though the two strings are not matched closely by difflib standards.

Reply:
Related, converting all strings to lower case will make it easier to compare between any of them.

Reply:
those are fantastic ideas-- thanks! I will implement them :D